### PR TITLE
Port TR_J9JITServerSharedCache from jitaas to master

### PR DESCRIPTION
**Pull request by harryyu1994**
TR_J9JITServerSharedCache is an extension to TR_J9SharedCache that overrides a few APIs from TR_J9SharedCache. This class is used by JITServer to remotely obtain shared-cache-related information from JITClient.

Signed-off-by: Harry Yu <harryyu1994@gmail.com>

### CODE REVIEW

**Comment by harryyu1994** [comment]
@mpirvu Hi Marius, this one is ready for review.

File: runtime/compiler/control/JITClientCompilationThread.cpp [hunk 1]
```diff
@@ -1419,7 +1434,7 @@ handleServerMessage(JITServer::ClientStream *client, TR_J9VM *fe, JITServer::Mes
          J9Method *j9method = NULL;
             {
             TR::VMAccessCriticalSection getResolvedHandleMethod(fe);
-            j9method = jitGetImproperInterfaceMethodFromCP(fe->vmThread(), mirror->cp(), cpIndex);
+            j9method = jitGetImproperInterfaceMethodFromCP(fe->vmThread(), mirror->cp(), cpIndex, NULL);
```
Comment (mpirvu) [inline runtime/compiler/control/JITClientCompilationThread.cpp]: I would like some re-assurance that NULL is the proper solution here. PR #6993 changed the way resolved methods are created I would like to make sure that this is reflected in the jitaas solution

**Comment by mpirvu** [comment]
I am going to merge this since the errors seen are intermittent and not due to this PR.

**Comment by harryyu1994** [comment]
Migrated to https://github.com/eclipse/openj9/pull/7461

**Pull request merged**